### PR TITLE
Improve design of instance group header

### PR DESCRIPTION
**Pull request by TayouVR**
This is really just a stopgap as that entire instance view should be redone, see some earlier efforts in #30 
But for the time being I think its okay to put a little effort in this

<details><summary>Before</summary>

image

</details>

<details><summary>Intermediary</summary>

image

</details>

<details><summary>After (edited)</summary>

image

</details>

### CODE REVIEW

**Comment by TayouVR** [comment]
also for feedback, if for some themes the arrow is too close to the left, I can shift it a little. Just tell me!

**Comment by leia-uwu** [comment]
yeah it needs to be shifted a bit to the left
also the click hitbox on the arrow is a bit janky and there's a lof of warnings being spammed :p

image

File: launcher/ui/instanceview/VisualGroup.cpp [hunk 1]
```diff
@@ -203,7 +204,8 @@ void VisualGroup::drawHeader(QPainter *painter, const QStyleOptionViewItem &opti
 
         painter->save();
         painter->setFont(font);
-        painter->drawText(textRect, Qt::AlignLeft | Qt::AlignVCenter, text);
+        painter->drawText(textRect, Qt::AlignLeft | Qt::AlignVCenter, text != "" ? text : QObject::tr("Ungrouped"));
```
Comment (Scrumplex) [inline launcher/ui/instanceview/VisualGroup.cpp]: for consistency in the codebase I would propose that you should use `QString::isEmpty()`

**Review by TheKodeToad — CHANGES_REQUESTED** [review]
code needs a bit of formatting regarding operators :p

File: launcher/ui/instanceview/VisualGroup.cpp [hunk 2]
```diff
@@ -112,183 +101,105 @@ QPair<int, int> VisualGroup::positionOf(const QModelIndex &index) const
...
 }
 
-VisualGroup::HitResults VisualGroup::hitScan(const QPoint &pos) const
+VisualGroup::HitResults VisualGroup::hitScan(const QPoint& pos) const
 {
     VisualGroup::HitResults results = VisualGroup::NoHit;
     int y_start = verticalPosition();
     int body_start = y_start + headerHeight();
-    int body_end = body_start + contentHeight() + 5; // FIXME: wtf is this 5?
+    int body_end = body_start + contentHeight() + 5;  // FIXME: wtf is this 5?
```
Comment (TheKodeToad) [inline launcher/ui/instanceview/VisualGroup.cpp]: could you remove the magic 5 here?

File: launcher/ui/instanceview/VisualGroup.cpp [hunk 3]
```diff
@@ -112,183 +101,105 @@ QPair<int, int> VisualGroup::positionOf(const QModelIndex &index) const
...
-    //END: checkboxy thing
+    // END: arrow
 
-    //BEGIN: text
+    // BEGIN: text
     {
-        QRect textRect(option.rect);
-        textRect.setTop(textRect.top() + 7);
-        textRect.setLeft(textRect.left() + 7 + fontMetrics.height() + 7);
+        painter->setRenderHint(QPainter::Antialiasing);
```
Comment (TheKodeToad) [inline launcher/ui/instanceview/VisualGroup.cpp]: If you move `setRenderHint` after `save` this probably isn't needed

Comment (TayouVR) [reply launcher/ui/instanceview/VisualGroup.cpp]: after reading the docs for save and restore I think actually this isn't needed at all.

**Comment by TheKodeToad** [comment]
I think the text should be clickable, and the triangle should be antialiased.
image
this is pretty rough!

**Review by getchoo — APPROVED** [review]
this looks great! i got no other suggestions really <3

File: launcher/ui/instanceview/VisualGroup.cpp [hunk 4]
```diff
@@ -321,7 +227,7 @@ int VisualGroup::contentHeight() const
 
 int VisualGroup::numRows() const
 {
-    return rows.size();
+    return (int)rows.size();
```
Comment (TheKodeToad) [inline launcher/ui/instanceview/VisualGroup.cpp:230]: What's this cast for?

**Review by TheKodeToad — APPROVED** [review]

File: launcher/ui/instanceview/VisualGroup.cpp (hunk 4 above)
Comment (Ryex) [reply launcher/ui/instanceview/VisualGroup.cpp:230]: warning suppression  on data loss from conversion form a larger type like size_t I suspect

Comment (TheKodeToad) [reply launcher/ui/instanceview/VisualGroup.cpp:230]: Qt uses int

Comment (TayouVR) [reply launcher/ui/instanceview/VisualGroup.cpp:230]: image

Comment (TheKodeToad) [reply launcher/ui/instanceview/VisualGroup.cpp:230]: Ah, i'm using Qt 5 for breeze lol

**Pull request merged**